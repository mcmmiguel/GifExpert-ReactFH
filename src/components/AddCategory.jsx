import { useState } from "react";

const AddCategory = () => {

    const [inputValue, setInputValue] = useState('One Punch');

    const onInputChange = (event) => {
        setInputValue(event.target.value);
        console.log(event.target.value);
    };

    const onSubmit = (event) => {
        event.preventDefault();
        console.log(event);
    };

    return (

        <form onSubmit={(event) => onSubmit(event)}>
            <input
                type="text"
                placeholder="Buscar GIF's"
                value={inputValue}
                onChange={onInputChange}
            />
        </form>

    );
};

export default AddCategory