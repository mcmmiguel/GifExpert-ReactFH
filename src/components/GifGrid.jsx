import { useState, useEffect } from "react";
import { getGifs } from "../helpers/getGifs";
import GifItem from "./GifItem";

const GifGrid = ({ category }) => {

    const [images, setImages] = useState([]);

    const getImages = async () => {
        const newImages = await getGifs(category);
        setImages(newImages);
        console.log(newImages);
    }

    useEffect(() => {
        getImages();
    }, []);


    return (
        <>
            <h3>{category}</h3>
            <div className="card-grid">
                {
                    images.map((img) => (
                        <GifItem
                            key={img.id}
                            {...img}
                        />
                    ))
                }
            </div>
        </>
    )
}

export default GifGrid