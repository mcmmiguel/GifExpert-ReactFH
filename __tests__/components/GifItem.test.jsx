/* eslint-disable no-undef */
import { render } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Pruebas en <GifItem />', () => {

    const title = 'One Punch Man GIF';
    const url = 'https://media2.giphy.com/media/VXJWhaO7afRe/giphy.gif?cid=35ddf05cuw50ny5k6ptqlxa7v60221zmuek58071unojpybk&ep=v1_gifs_search&rid=giphy.gif&ct=g';

    test('Debe hacer match con el snapshot', () => {

        const { container } = render(<GifItem title={title} url={url} />);
        expect(container).toMatchSnapshot();

    });

})