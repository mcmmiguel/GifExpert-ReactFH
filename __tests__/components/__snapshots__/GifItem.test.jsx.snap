// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="One Punch Man GIF"
      src="https://media2.giphy.com/media/VXJWhaO7afRe/giphy.gif?cid=35ddf05cuw50ny5k6ptqlxa7v60221zmuek58071unojpybk&ep=v1_gifs_search&rid=giphy.gif&ct=g"
    />
    <p>
      One Punch Man GIF
    </p>
  </div>
</div>
`;
